### ISSUE TITLE
Nested Arguments on Fields within a ListType not working properly

### ISSUE THREAD

**Issue by jasonbahl** [Youshido/GraphQL#111]
I initially posted a comment over here (https://github.com/Youshido/GraphQL/issues/109#issuecomment-275881911), but it seems like it might make sense to have a different Issue as they may or may not be related. 

It seems that arguments are getting cleared on fields that are nested within a ListType after the first item of the List is processed.

See the following query and results. You'll see the issue occurring with the `postsConnection` field's args. 

I pass the args `args: {per_page: 1}` and in the query results, you can see the raw output of the `raw_args` field (which is a field I added that simply captures the `$args` from the resolve method and outputs it as a string). You can see the first pass appropriately shows the input of `per_page: 1` coming through, but the second output shows no args coming through at all, so it seems that the $args are getting cleared at some point.

```
query{
  posts(args:{
    perPage: 2
  }){
    items{
      id
      title
      link
      categoriesConnection( args: {
        per_page: 3
      }){
        items{
          id
          count
          postsConnection(args: {
            per_page: 1
          }){
            raw_args
            processed_args
            items{
              id
              title
            }
          }
        }
      }
    }
  }
}
```

And the results that are generated: 

```
{
  "data": {
    "posts": {
      "items": [
        {
          "id": 2223151,
          "title": "ES Test 2",
          "link": "http://localsite.dev/2017/01/10/es-test-2/",
          "categoriesConnection": {
            "items": [
              {
                "id": 16,
                "count": 252,
                "postsConnection": {
                  "raw_args": "{\"args\":{\"per_page\":1},\"posts_per_page\":1,\"post_type\":\"post\"}",
                  "processed_args": "{\"per_page\":1}",
                  "items": [
                    {
                      "id": 2223151,
                      "title": "ES Test 2"
                    }
                  ]
                }
              },
              {
                "id": 104,
                "count": 130,
                "postsConnection": {
                  "raw_args": "{\"args\":[],\"posts_per_page\":5,\"post_type\":\"post\"}",
                  "processed_args": "[]",
                  "items": [
                    {
                      "id": 2223151,
                      "title": "ES Test 2"
                    },
                    {
                      "id": 2223148,
                      "title": "ES Test"
                    },
                    {
                      "id": 2162701,
                      "title": "1"
                    },
                    {
                      "id": 2223124,
                      "title": "Some cool Title"
                    },
                    {
                      "id": 2223114,
                      "title": "Some cool Title"
                    }
                  ]
                }
              },
              {
                "id": 17,
                "count": 840,
                "postsConnection": {
                  "raw_args": "{\"args\":[],\"posts_per_page\":5,\"post_type\":\"post\"}",
                  "processed_args": "[]",
                  "items": [
                    {
                      "id": 2223151,
                      "title": "ES Test 2"
                    },
                    {
                      "id": 2223148,
                      "title": "ES Test"
                    },
                    {
                      "id": 2162701,
                      "title": "1"
                    },
                    {
                      "id": 2223124,
                      "title": "Some cool Title"
                    },
                    {
                      "id": 2223114,
                      "title": "Some cool Title"
                    }
                  ]
                }
              }
            ]
          }
        },
        {
          "id": 2223148,
          "title": "ES Test",
          "link": "http://localsite.dev/2017/01/09/es-test/",
          "categoriesConnection": {
            "items": [
              {
                "id": 16,
                "count": 252,
                "postsConnection": {
                  "raw_args": "{\"args\":[],\"posts_per_page\":5,\"post_type\":\"post\"}",
                  "processed_args": "[]",
                  "items": [
                    {
                      "id": 2223151,
                      "title": "ES Test 2"
                    },
                    {
                      "id": 2223148,
                      "title": "ES Test"
                    },
                    {
                      "id": 2162701,
                      "title": "1"
                    },
                    {
                      "id": 2223124,
                      "title": "Some cool Title"
                    },
                    {
                      "id": 2223114,
                      "title": "Some cool Title"
                    }
                  ]
                }
              }
            ]
          }
        }
      ]
    }
  }
}
```

I'll dig more and see what I can find out, but thought these details might help point out either an issue with the code, or an issue with my implementation. 

Thanks!

**Comment by jasonbahl** [comment]
After digging a bit more, it appears that the issue is only occurring with args of the InputObjectType. 

I structured some of my args to be nested, but if I flatten all my args to be on the same level, things appear to be working. 

I still do have some fields that require nested input, and they seem to be having the issue still, but I've moved the bulk of my $args up to the top-level and they seem to be working now.

From what I can tell, it might be an issue in the Parser.php file not appropriately parsing the nested input objects, but I wasn't able to come to any solid conclusion on what the issue is.
